src/SceneManager.ts

```diff
@@ -199,6 +243,13 @@ export class SceneManager {
     }
   }
 
+  public publish(): void {
+    if (this.scene) {
```

Comment:
Check for the publisher as well.

```suggestion
    if (this.scene && this.publisher) {
```